internal/clients/manager.go

```diff
@@ -170,7 +206,37 @@ func (c *AdminAPIClientsManager) KonnectClient() *adminapi.KonnectClient {
 func (c *AdminAPIClientsManager) GatewayClients() []*adminapi.Client {
 	c.lock.RLock()
 	defer c.lock.RUnlock()
-	return lo.Values(c.readyGatewayClients)
+	readyGatewayClients := lo.Values(c.readyGatewayClients)
+	// With dbless mode, we should send configuration to ALL Kong gateway instances.
+	if dataplaneutil.IsDBLessMode(c.dbMode) {
+		return readyGatewayClients
+	}
+	// if there are no ready gateway clients, we should return an empty list.
+	if len(readyGatewayClients) == 0 {
+		return []*adminapi.Client{}
+	}
+	// return ONE client if Kong gateway is DB backed.
+	// if the name of admin API service is given, return the address of the service.
+	if c.adminAPIServiceName.Name != "" && c.adminAPIServiceName.Namespace != "" {
+		cl, err := c.clientFactory.CreateAdminAPIClient(c.ctx, adminapi.DiscoveredAdminAPI{
+			Address: fmt.Sprintf("https://%s.%s:%d",
+				c.adminAPIServiceName.Name,
+				c.adminAPIServiceName.Namespace,
+				c.adminAPIServicePort,
+			),
+		})
+		if err != nil {
+			// Fallback to choose a single instance if we failed to create a client from admin API service.
+			c.logger.Error(err, "failed to create admin API client from admin API service address",
+				"service_name", c.adminAPIServiceName.String(),
+				"service_port", c.adminAPIServicePort,
+			)
+		} else {
+			return []*adminapi.Client{cl}
+		}
+	}
+	// if no admin API service is given, choose one discovered client to send configurations.
+	return readyGatewayClients[:1]
```

Comment:
I think this should not happen on every `GatewayClients()` call. I'd suggest that in the case of DBMode, the `initialClients` passed to `NewAdminAPIClientsManager` constructor should have one element with a pre-created client connected to the admin API service. We should also not call `AdminAPICLientsManager.Run()` in such case because it has no job to do - there are no readiness checks to perform and no discovered Admin APIs to handle.

Reply:
removed this because we decided to return one single client.